addons/web/static/tests/core/dropdown/dropdown.test.js

```diff
@@ -69,6 +74,24 @@ class MultiLevelDropdown extends Component {
     `;
 }
 
+class DropdownInShadowDom extends Component {
+    static components = { SimpleDropdown };
+    static props = [];
+    static template = xml`<div><SimpleDropdown/></div>`;
+}
+class ShadowDom extends Component {
+    static components = { Dropdown, DropdownItem };
+    static props = [];
+    static template = xml`<div class="shadow-root" />`;
+    setup() {
+        onMounted(() => {
+            const shadowRoot = document.querySelector(".shadow-root");
```

Comment:
With components, try use refs instead of document.querySelector as it's more robust, it avoids any error if the same class name is uses somewhere else
```js
const shadowRef = useRef("shadow-root");
onMounted(() => {
    const shadowRoot = shadowRef.el;
```